test/php/Nominatim/ParameterParserTest.php

```diff
@@ -0,0 +1,214 @@
+
+        try {
+            $this->assertEquals(false, $oParams->getSet('val2', ['foo', 'bar']));
+        } catch (Exception $e) {
+            $this->assertEquals($e->getMessage(), "Parameter 'val2' must be one of: foo, bar");
+        }
+        $this->assertEquals(false, $oParams->getSet('val3', ['foo', 'bar']));
+        $this->assertEquals(false, $oParams->getSet('val4', ['foo', 'bar']));
+    }
+
+
+    public function testGetStringList()
+    {
+        $oParams = new ParameterParser([
+                                        'list1' => ',a,b,c,,c,d',
+                                        'list2' => 'a',
+                                        'list3' => '',
+                                        'list4' => '0'
+                                       ]);
+
+        $this->assertEquals(false, $oParams->getStringList('non-exists'));
+        $this->assertEquals(['a', 'b'], $oParams->getStringList('non-exists', ['a', 'b']));
+        // FIXME: unclear if empty string items should be removed
+        $this->assertEquals(['', 'a', 'b', 'c', '', 'c', 'd'], $oParams->getStringList('list1'));
+        $this->assertEquals(['a'], $oParams->getStringList('list2'));
+        $this->assertEquals(false, $oParams->getStringList('list3'));
+        $this->assertEquals(false, $oParams->getStringList('list4'));
+    }
+
+    public function testGetPreferredLanguages()
+    {
+        $oParams = new ParameterParser(['accept-language' => '']);
+        $this->assertEquals([
+                             'brand' => 'brand',
+                             'ref' => 'ref',
+                             'type' => 'type',
+                             'name' => 'name',
+                             'name:default' => 'name:default',
+                             'short_name' => 'short_name',
+                             'short_name:default' => 'short_name:default',
+                             'official_name' => 'official_name',
+                             'official_name:default' => 'official_name:default',
+                            ], $oParams->getPreferredLanguages('default'));
+
+        $oParams = new ParameterParser(['accept-language' => 'de,en']);
+        $this->assertEquals([
+                             'brand' => 'brand',
+                             'ref' => 'ref',
+                             'type' => 'type',
+                             'name' => 'name',
+                             'name:de' => 'name:de',
+                             'name:en' => 'name:en',
+                             'short_name' => 'short_name',
+                             'short_name:de' => 'short_name:de',
+                             'short_name:en' => 'short_name:en',
+                             'official_name' => 'official_name',
+                             'official_name:de' => 'official_name:de',
+                             'official_name:en' => 'official_name:en',
+                            ], $oParams->getPreferredLanguages('default'));
```

Comment:
This is surprising. I would have expected name:xx and short_name:XX to come before brand and ref.

Reply:
It looks like assertEquals() does not take order into account for arrays but order actually matters for the results here.